kyo-direct/shared/src/main/scala/kyo/Validate.scala

```diff
@@ -126,7 +147,7 @@ private[kyo] object Validate:
                     |}""".stripMargin)}""".stripMargin.stripMargin
                 )
 
-            case tree @ DefDef(_, _, _, Some(body)) if !pure(body) =>
+            case tree @ DefDef(_, _, _, Some(body)) if !pure(body) && !validShift(body) =>
```

Comment:
If dotty-cps-async provides reasonable error messages for this scenario, another option is just removing the restriction. It'd be good if we can avoid the tree pattern matching

Reply:
Ha, I see. The updated implementation if doing something else, just jump to part of the tree to check.

https://github.com/getkyo/kyo/pull/1197/files#diff-4f91081931ccb84da7b96c5f6764fae9e5c84903e1e521379f51dad387dd7e73R97-R99

We could remove the restriction, however there are constructions that are valid for `dotty-cps-async`, and not encouraged in the context of Kyo (using var, ...).